htlcswitch/test_utils.go

```diff
@@ -384,6 +384,21 @@ func generateHops(payAmt btcutil.Amount,
 	return runningAmt, totalTimelock, hops
 }
 
+type paymentResponse struct {
+	invoice *channeldb.Invoice
+	err     chan error
+}
+
+func (r *paymentResponse) Wait(d time.Duration) (*channeldb.Invoice, error) {
+	select {
+	case err := <-r.err:
+		close(r.err)
```

Comment:
Why close the error channel? This will cause any subsequent sends to cause the entire test to panic.